ingest/scripts/get_loculus_submissions.py

```diff
@@ -0,0 +1,182 @@
+    errors: str | None = None
+    warnings: str | None = None
+    status: Status = Status.READY
+    started_at: datetime | None = None
+    finished_at: datetime | None = None
+    result: str | None = None
+
+
+@dataclass
+class AssemblyTableEntry:
+    accession: str
+    version: int
+    errors: str | None = None
+    warnings: str | None = None
+    status: Status = Status.READY
+    started_at: datetime | None = None
+    finished_at: datetime | None = None
+    result: str | None = None
+
+
+def get_bio_sample_accessions(db_conn_pool: SimpleConnectionPool) -> dict[str, str]:
+    con = db_conn_pool.getconn()
+    try:
+        with con, con.cursor(cursor_factory=RealDictCursor) as cur:
+            # Result is a jsonb column
+            query = "SELECT accession, result FROM sample_table WHERE STATUS = 'SUBMITTED'"
+
+            cur.execute(query)
+
+            results = cur.fetchall()
+    finally:
+        db_conn_pool.putconn(con)
+
+    return {result["accession"]: result["result"]["biosample_accession"] for result in results}
+
+
+def get_insdc_accessions(db_conn_pool: SimpleConnectionPool) -> dict[str, str]:
+    con = db_conn_pool.getconn()
+    try:
+        with con, con.cursor(cursor_factory=RealDictCursor) as cur:
+            # Result is a jsonb column
+            query = "SELECT accession, result FROM assembly_table WHERE STATUS = 'SUBMITTED'"
+
+            cur.execute(query)
+
+            results = cur.fetchall()
+    finally:
+        db_conn_pool.putconn(con)
+
+    return {
+        result["accession"]: [
+            result["result"][key]
+            for key in result["result"]
+            if key.startswith("insdc_accession_full")
+        ]
+        for result in results
+    }
+
+
+@click.command()
+@click.option(
+    "--log-level",
+    default="INFO",
+    type=click.Choice(["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]),
+)
+@click.option(
+    "--config-file",
+    required=True,
+    type=click.Path(exists=True),
+)
+@click.option(
+    "--output-insdc-accessions",
+    required=True,
+    type=click.Path(),
+)
+@click.option(
+    "--output-biosample-accessions",
+    required=True,
+    type=click.Path(),
+)
+def filter_out_loculus_submissions(
+    log_level, config_file, output_insdc_accessions, output_biosample_accessions
+):
+    logger.setLevel(log_level)
+    logging.getLogger("requests").setLevel(logging.INFO)
+
+    with open(config_file, encoding="utf-8") as file:
+        full_config = yaml.safe_load(file)
+        relevant_config = {key: full_config.get(key, []) for key in Config.__annotations__}
+        config = Config(**relevant_config)
+    logger.info(f"Config: {config}")
+
+    db_config = db_init(config.db_password, config.db_username, config.db_host)
+    insdc_accessions_submitted_by_loculus = get_insdc_accessions(db_config)
+    all_insdc_accessions_submitted_by_loculus = [
+        item for sublist in insdc_accessions_submitted_by_loculus.values() for item in sublist
```

Comment:
I remember at Google the linter didn't allow nested list comprehensions because they are too hard to understand - IMO there's a reasonable argument to avoid them but not blocking.